### PR TITLE
Feature/KChipGroup

### PR DESCRIPTION
**Pull request by michaelbukachi**

### CODE REVIEW

File: kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupActions.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import android.view.View
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom
+import com.agoda.kakao.common.actions.BaseActions
+import com.google.android.material.chip.Chip
+import com.google.android.material.chip.ChipGroup
+
+interface ChipGroupActions : BaseActions {
```
Comment (Unlimity) [inline kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupActions.kt]: I think it needs `selectChip(@IdRes id: Int)` and `selectChipAt(index: Int)`

File: kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupAssertions.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    fun hasChip(matcher: Matcher<String>) {
+        view.check(ViewAssertions.matches(ViewMatchers.hasDescendant(withText(matcher))))
+    }
+
+    /**
+     * Check if Chip with matching text in ChipGroup is selected
+     *
+     * @param text text in chip
+     */
+    fun chipIsSelected(text: String) {
```
Comment (Unlimity) [inline kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupAssertions.kt]: Naming conventions suggests that it should be `isChipSelected`

File: kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupAssertions.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.agoda.kakao.common.assertions.BaseAssertions
+import com.agoda.kakao.common.matchers.SelectedChipMatcher
+import org.hamcrest.Matcher
+
+/**
+ * Provides assertions for a ChipGroup
+ */
+interface ChipGroupAssertions : BaseAssertions {
```
Comment (Unlimity) [inline kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupAssertions.kt]: Would be helpful to add `hasSize(size: Int)` and `hasChip(@IdRes id: Int)` assertions.

Comment (Unlimity) [reply kakao/src/main/kotlin/com/agoda/kakao/chipgroup/ChipGroupAssertions.kt]: And `isChipSelected(matcher: Matcher<String>)` + `isChipSelected(@IdRes id: Int)`

File: sample/src/androidTest/kotlin/com/agoda/sample/ChipGroupTest.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @JvmField
+    val rule = ActivityTestRule(ChipGroupActivity::class.java)
+
+    @Test
+    fun testChipGroupInteraction() {
+        onScreen<ChipGroupActivityScreen> {
+            chipGroup {
+                selectChip("Chip2")
+
+                idle(5000)
```
Comment (Unlimity) [inline sample/src/androidTest/kotlin/com/agoda/sample/ChipGroupTest.kt]: Please do not idle for that long. 500-1000 ms is more than enough. But actually, there should not be any idle at all.

Comment (michaelbukachi) [reply sample/src/androidTest/kotlin/com/agoda/sample/ChipGroupTest.kt]: Whoa. I totally forgot to about that line. Was using it for testing. Definitely not supposed to be there :sweat_smile:

**Comment by michaelbukachi** [comment]
@Unlimity do you mind reviewing the changes?

**Comment by Vacxe** [comment]
LGTM

**Comment by Vacxe** [comment]
Close: #194

**Comment by Vacxe** [comment]
Thanks for the contribution

**Pull request merged**